images/administerkmskeys_policy.tf

```diff
@@ -0,0 +1,33 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policy that allows all of the KMS actions necessary to
+# administer KMS keys in the Images account
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "administerkmskeys_doc" {
+  statement {
+    actions = [
+      "kms:CancelKeyDeletion",
+      "kms:Create*",
+      "kms:Delete*",
+      "kms:Describe*",
+      "kms:Disable*",
+      "kms:Enable*",
+      "kms:Get*",
+      "kms:List*",
+      "kms:Put*",
+      "kms:Revoke*",
+      "kms:ScheduleKeyDeletion",
+      "kms:TagResource",
+      "kms:UntagResource",
+      "kms:Update*",
+    ]
+
+    resources = ["*"]
```

Comment:
Do we want a policy for administering all KMS keys, or only the KMS key used to encrypt AMIs?

Reply:
KMS keys appear to have unique IDs with no discernible pattern (as far as I can tell), so in order to create/modify/destroy any KMS keys, we need a policy allowing administration of all keys.

BTW, I found this doc about [AWS KMS Best Practices](https://d0.awsstatic.com/whitepapers/aws-kms-best-practices.pdf), which I need to read a bit more thoroughly, but wanted to share in case anyone else wants to take a gander at it.

Reply:
I will take a look at that document in the morning.